File: 0506-relative-ranks/0506-relative-ranks.py
class Solution:
	def findRelativeRanks(self, score: List[int]) -> List[str]:

		result = []
		hashmap = defaultdict(int)
		rank = sorted(score , reverse = True)

		for index in range(len(rank)):

			hashmap[rank[index]] = index

		for num in score:

			if hashmap[num] == 0:
				result.append('Gold Medal')
			elif hashmap[num] == 1:
				result.append('Silver Medal')
			elif hashmap[num] == 2:
				result.append('Bronze Medal')
			else:
				result.append(str(hashmap[num] + 1))

		return result

File: 0506-relative-ranks/0506-relative-ranks.cpp
class Solution {
public:
    vector<string> findRelativeRanks(vector<int>& score) {
        priority_queue<pair<int,int>> pq;
        int i=0;
        for(auto &it: score) pq.push({it,i++});
        vector<string> ans(score.size());
        if(pq.size()>0){ ans[pq.top().second] = "Gold Medal"; pq.pop();}
        if(pq.size()>0){ ans[pq.top().second] = "Silver Medal"; pq.pop();}
        if(pq.size()>0){ ans[pq.top().second] = "Bronze Medal"; pq.pop();}
        int r = 4;
        while(!pq.empty()){
            ans[pq.top().second] = to_string(r++);
            pq.pop();
        }
        return ans;
    }
};